README.md

```diff
@@ -89,6 +89,42 @@ input = ["KEY_F8"]
 output = ["KEY_MUTE"]
 ```
 
+`REL_*` and `ABS_*` input types are also supported. These have an associated 
+positive or negative value depending on the relative direction of the input.
+You can optionally specify the relative direction by appending a "+" or "-" 
+to the input name. For example, if you wanted to remap your scroll wheel to 
+a keypress:
+
+```toml
+[[remap]]
+input = ["REL_WHEEL+"]
+output = ["KEY_UP"]
+
+[[remap]]
+input = ["REL_WHEEL-"]
+output = ["KEY_DOWN"]
+```
+
+Mapping relative inputs to relative inputs is also possible, you can include 
+a scaling factor to both the input value and output. The input value is divided 
+by the input scaling factor, and the output is multiplied by the output scaling 
+factor. Specify a negative scale factor on the output to invert the input. If you 
+provide an input scale, you need to remap both relative directions separately:
```

Comment:
This is quite dense and is hard to immediately grok and relate to the examples below.
Please intersperse this explanation with the examples and explain precisely what happens for each of the examples, as it isn't immediately clear what the `+4` on input does in relation to the `+2` on output.

Reply:
FWIW, I think the scale factor looks like addition/subtraction, which makes it a bit confusing, as does the context dependent multiply-or-divide.

How about using something like `+WHEEL/4` to indicate positive relative wheel, divided by 4, or `-WHEEL*2` to indicate negative relative wheel multiplied by 2?

I think switching the scale from `i32` to an `f32` is conceptually good for this, as the parser can more directly track the intent that was written out in the config file. I think the wrinkle would be that `f32` doesn't support `Eq` (only `PartialEq`). That could be resolved by using [NotNan<f32>](https://docs.rs/ordered-float/4.2.0/ordered_float/struct.NotNan.html).

With that approach, the default scale would be `1.0` and the other scaling logic is simply the product of both input and output scale factors.